src/CommandGenerator/CommandGenerator.cs

```diff
@@ -129,16 +129,12 @@ public DbCommand GetUpdateCommand()
         /// <returns>
         /// Upsert command.
         /// </returns>
-        public DbCommand GetUpsertCommand()
+        public DbCommand GetUpsertCommand(IEnumerable<string> conditionColumnNames = null)
         {
-            ThrowHelper.CheckAndThrowMethodNotSupportedWhenNoPrimaryKey(_tableInfo);
-
-            IEnumerable<ColumnInfo> columns = GetQueryColumns(ValueGenerated.OnUpdate);
-            DbCommand cmd = _provider.GetCommandForCurrentTransaction();
-            AddParametersToCommand(cmd, columns.Where(x => !x.IsPrimaryKey));
-            AddParametersToCommand(cmd, columns.Where(x => x.IsPrimaryKey));
-            cmd.CommandText = GetUpsertCommandText();
-            return cmd;
+            IEnumerable<ColumnInfo> conditonColumns = conditionColumnNames?.Any() == true
```

Comment:
Chýba mi tu nejaká kontrola, či naozaj mám stĺpce pre podmienku. V prípade, že na vstupe v `conditionColumnNames` pošlem blbosti, tak `conditonColumns` bude prázdne `IEnumerable`. To potom v `GetUpsertCommandText` vygeneruje blbé SQL-ko a ťažko sa to bude dohľadávať.